### PR TITLE
Kafka exporter

### PR DESCRIPTION
**Pull request by prestonvanloon**
Exports JSON copies of attestations and beacon blocks as they are saved to the database. 

Run with `--kafka-url=localhost:9092` to connect to a local kafka server. 

Observations / thoughts:
- This adds about 3.5% overhead when syncing.
- This only streams objects that are saved to the database, attestations / blocks that cannot be processed may not be streamed
- Aggregated attestations may not be properly streamed, need to confirm this

### CODE REVIEW

File: beacon-chain/db/db.go [hunk 1]
```diff
@@ -1,95 +1,21 @@
...
 
 // NewDB initializes a new DB.
 func NewDB(dirPath string) (Database, error) {
-	return kv.NewKVStore(dirPath)
+	db, err := kv.NewKVStore(dirPath)
+	if err != nil {
+		return nil, err
+	}
+
+	return kafka.Wrap(db)
```
Comment (nisdas) [inline beacon-chain/db/db.go]: Shouldnt this be configured using the feature flag ?

Comment (prestonvanloon) [reply beacon-chain/db/db.go]: It is, see implementation of kafka.Wrap

Comment (prestonvanloon) [reply beacon-chain/db/db.go]: maybe it should be moved out here too

File: beacon-chain/db/kafka/export_wrapper.go [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+	}()
+
+	return e.db.SaveAttestation(ctx, att)
+}
+
+// SaveAttestations publishes to the kafka topic for beacon attestations.
+func (e Exporter) SaveAttestations(ctx context.Context, atts []*eth.Attestation) error {
+	go func() {
+		for _, att := range atts {
+			if err := e.publish(ctx, "beacon_attestation", att); err != nil {
```
Comment (shayzluf) [inline beacon-chain/db/kafka/export_wrapper.go]: topic should be the same as above? or same as here?
```suggestion
			if err := e.publish(ctx, "attestation", att); err != nil {
```

File: beacon-chain/db/kafka/export_wrapper.go [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+	}()
+
+	return e.db.SaveBlock(ctx, block)
+}
+
+// SaveBlocks publishes to the kafka topic for beacon blocks.
+func (e Exporter) SaveBlocks(ctx context.Context, blocks []*eth.BeaconBlock) error {
+	go func() {
+		for _, block := range blocks {
+			if err := e.publish(ctx, "block", block); err != nil {
```
Comment (shayzluf) [inline beacon-chain/db/kafka/export_wrapper.go]: same as above

File: beacon-chain/db/kafka/export_wrapper.go [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+			log.WithError(err).Error("Failed to publish block")
+		}
+	}()
+
+	return e.db.SaveBlock(ctx, block)
+}
+
+// SaveBlocks publishes to the kafka topic for beacon blocks.
+func (e Exporter) SaveBlocks(ctx context.Context, blocks []*eth.BeaconBlock) error {
+	go func() {
```
Comment (shayzluf) [inline beacon-chain/db/kafka/export_wrapper.go]: maybe we can create a go function for each publish?

**Comment by prestonvanloon** [comment]
Will finish this after the holiday.

File: beacon-chain/db/kafka/export_wrapper.go (hunk 3 above)
Comment (prestonvanloon) [reply beacon-chain/db/kafka/export_wrapper.go]: Yeah, this should be `beacon_block`

File: beacon-chain/db/kafka/export_wrapper.go (hunk 4 above)
Comment (prestonvanloon) [reply beacon-chain/db/kafka/export_wrapper.go]: I think this is OK. It's already non-blocking in the main routine, i think adding more routines might create unnecessary overhead.

File: beacon-chain/db/kafka/export_wrapper.go (hunk 2 above)
Comment (prestonvanloon) [reply beacon-chain/db/kafka/export_wrapper.go]: Resolved by using `beacon_attestation`, nice catch!

**Pull request merged**